src/components/deal-proposal-form/DealProposalForm.tsx

```
@@ -100,20 +105,51 @@ export function DealProposalForm({
   });
 
   const [startBlock, endBlock, pricePerBlock] = watch(["startBlock", "endBlock", "pricePerBlock"]);
-
   const duration = endBlock - startBlock;
   const startBlockRealTime = blockToTime(startBlock, currentBlock, currentBlockTimestamp);
   const endBlockRealTime = blockToTime(endBlock, currentBlock, currentBlockTimestamp);
   const totalPrice = duration * pricePerBlock;
 
+  const { collatorWsApi: api } = useCtx();
+  const client = watch("client");
+  const [marketBalance, setMarketBalance] = useState<number>(0);
+  const [balanceStatus, setBalanceStatus] = useState<MarketBalanceStatus>(MarketBalanceStatus.Idle);
+
+  const fetchMarketBalance = useCallback(async () => {
+    if (!api || !client) return;
```

This should throw an error but let's ship as is rn